platform/security/sts/security-sts-propertyclaimshandler/src/test/java/org/codice/ddf/security/sts/claims/property/TestAttributeFileClaimsHandler.java

```diff
@@ -44,120 +44,315 @@
+    System.setProperty(SystemBaseUrl.HOST, "someUserThatIsntInTheUsersDotAttributesFile");
+
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(getPathForValidTestAttributesFile());
+    attributeFileClaimsHandler.init();
+
+    // when
+    attributeFileClaimsHandler.init();
   }
 
-  @Test
-  public void testRetrieveClaimsValuesNullPrincipal() {
-    ClaimsParameters claimsParameters = new ClaimsParameters();
-    ClaimCollection claimCollection = new ClaimCollection();
-    ProcessedClaimCollection processedClaims =
-        attributeFileClaimsHandler.retrieveClaimValues(claimCollection, claimsParameters);
+  @Test(expected = NullPointerException.class)
+  public void testSetNullAttributeFileLocation() {
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(null);
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testNoUsersDotAttributesFile() {
+    // given
+    System.setProperty(SystemBaseUrl.HOST, "user1");
+
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation("noFileHere");
+    attributeFileClaimsHandler.init();
+
+    // when
+    attributeFileClaimsHandler.init();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testUnexpectedFormatInUsersDotAttributesFile() throws IOException {
+    // given
+    System.setProperty(SystemBaseUrl.HOST, "user1");
+
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(
+        createAttributeFilePathFromResourceFileName("users.attributes-unexpectedFormat"));
+    attributeFileClaimsHandler.init();
+
+    // when
+    attributeFileClaimsHandler.init();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testEmptyListValueInUsersDotAttributesFile() throws IOException {
+    // given
+    System.setProperty(SystemBaseUrl.HOST, "user1");
+
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(
+        createAttributeFilePathFromResourceFileName("users.attributes-emptyListValue"));
+    attributeFileClaimsHandler.init();
+
+    // when
+    attributeFileClaimsHandler.init();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testNonStringValueInUsersDotAttributesFile() throws IOException {
+    // given
+    System.setProperty(SystemBaseUrl.HOST, "user1");
+
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(
+        createAttributeFilePathFromResourceFileName("users.attributes-nonStringValue"));
```

Comment:
✏️  Make this filename static final variables and re-use.

Reply:
Each of the resource files is only used in one unit test method, except for the valid "users.attributes", which is only specified in one helper method. I don't think moving these filenames to static Strings would provide any value.